packages/ui/src/components/organisms/DynamicForm/components/RSJVInputAdaters/DateInputAdater/DateInputAdater.tsx

```diff
@@ -1,17 +1,47 @@
-import { DatePickerInput } from '@/components/molecules';
+import { DatePickerChangeEvent, DatePickerInput } from '@/components/molecules';
 import { RJSFInputAdapter } from '@/components/organisms/DynamicForm/components/RSJVInputAdaters/types';
 import { useCallback } from 'react';
 
-export const DateInputAdater: RJSFInputAdapter = ({ id, formData, disabled, onBlur, onChange }) => {
+export const isValidDate = (dateString: string): boolean => {
+  const date = new Date(dateString);
+  if (date.getFullYear() < 1000) return false;
+
+  return true;
+};
+
+export const DateInputAdater: RJSFInputAdapter<string | null> = ({
+  id,
+  formData,
+  disabled,
+  onBlur,
+  onChange,
+}) => {
   const handleBlur = useCallback(() => {
     // @ts-ignore
     onBlur && onBlur(id, formData);
   }, [id, onBlur, formData]);
 
+  const handleChange = useCallback(
+    (event: DatePickerChangeEvent) => {
+      const dateValue = event.target.value;
+      if (dateValue === null) return onChange(null);
+
+      const dateString = dateValue;
+
+      // every onChange call in context of DateInput forces re-render of component which eventually dicsonnects user focus from input
+      // in cases when date considered as valid e.g 05/12/0100 but doesnt make sense onChange call should be skipped
+      // validating date with helper below
+      if (!isValidDate(dateString)) return;
+
+      onChange(dateString);
```

Comment:
TypeScript should be enough for where naming is lacking, re-assignment raises questions and adds unintentional meaning.
```suggestion
      const dateValue = event.target.value;
      
      if (dateValue === null) return onChange(null);

      // every onChange call in context of DateInput forces re-render of component which eventually dicsonnects user focus from input
      // in cases when date considered as valid e.g 05/12/0100 but doesnt make sense onChange call should be skipped
      // validating date with helper below
      if (!isValidDate(dateValue)) return;

      onChange(dateValue);
```